packages/chrome-extension/src/App.tsx

```diff
@@ -54,58 +70,54 @@ function useRscMessages() {
   const [isRecording, setIsRecording] = useState(false);
   const tabId = useMemo(() => Date.now(), []);
 
-  useEffect(() => {
-    function handleMessage(
-      request: unknown,
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      _sender: unknown,
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      _sendResponse: unknown,
-    ) {
-      if (
-        !isRscChunkMessage(request) &&
-        !isContentScriptUnloadedMessage(request)
-      ) {
-        return true;
-      }
-
-      // If the message is from a different tab, ignore it
-      if (request.tabId !== tabId) {
-        return true;
-      }
-
-      if (isContentScriptUnloadedMessage(request)) {
-        setIsRecording(false);
-        setMessages([]);
-        return true;
-      }
-
-      // It's possible that this lookup will miss a duplicated message if another
-      // one is being added at the same time. I haven't seen this happen in practice.
-      if (
-        messages.some((item) =>
-          arraysEqual(item.data.chunkValue, request.data.chunkValue),
-        )
-      ) {
-        return true;
-      }
-
-      startTransition(() => {
-        setMessages((previous) => [...previous, request]);
+  const handleNetworkRequest = useCallback(function (
+    req: chrome.devtools.network.Request,
+  ) {
+    const request = req as NetworkPanelRequest;
+    if (isRscResponse(request)) {
+      request.getContent((content) => {
+        const startTime = new Date(request.startedDateTime).getTime();
+
+        const requestMessage = {
+          fetchUrl: request.request.url,
+          fetchMethod: request.request.method,
+          fetchStartTime: startTime,
+          chunkStartTime: startTime,
+          chunkEndTime: startTime + request.time,
+          chunkValue: [],
+        } satisfies RscChunkMessage;
+
+        // there is an issue in chrome devtools, that during dev mode of nextjs
+        // the content is not available (it is also not shown within the network response tab)
+        if (content) {
+          requestMessage.chunkValue = Array.from(
+            new TextEncoder().encode(content),
+          );
+
+          startTransition(() => {
+            setMessages((previous) => [
+              ...previous,
+              { type: "RSC_CHUNK", tabId: 0, data: requestMessage },
+            ]);
+          });
+        }
       });
-
-      return true;
     }
+  }, []);
 
-    chrome.runtime.onMessage.addListener(handleMessage);
-
+  useEffect(() => {
     return () => {
-      chrome.runtime.onMessage.removeListener(handleMessage);
+      chrome.devtools.network.onRequestFinished.removeListener(
+        handleNetworkRequest,
+      );
     };
   }, []);
 
   const startRecording = useCallback(() => {
     setIsRecording(true);
+
+    chrome.devtools.network.onRequestFinished.addListener(handleNetworkRequest);
+
```

Comment:
`onRequestFinished` implies the full response has been returned right? The extension needs to be able to support HTTP streaming, so that you can later rewind time to see when specific chunks come in.

Do you know of a way to do that without using a fetch patch?

Reply:
Yes that is correct. You get access to the request and response. I was not aware of the timing feature for individual chunks within the same response. I guess that that is not possible with the API offered by the devtools. I will dig a little deeper to find a definitive answer for it. 

If it turns out that it is not possible, you can close the PR.

Reply:
You'd be able to see the future in action if you have a suspense boundary on a page with an artificially slowed down component inside. The response is a HTTP stream, and it won't wait for the slowed down component to finish to begin sending. 

> I will dig a little deeper to find a definitive answer for it.
Thanks, let me know what you find!